### ISSUE TITLE
nvcc fatal : Unsupported gpu architecture 'compute_75'

### ISSUE THREAD

**Issue by Morgan-117** [onnx/onnx-tensorrt#107]
I have some problems in "make -j8"
The cmake command works well and returns as following:
---------------------------------------------------------------------------------
-- The CXX compiler identification is GNU 5.4.0
-- The C compiler identification is GNU 5.4.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found CUDA: /usr/local/cuda (found version "9.0") 
-- Found Protobuf: /usr/local/lib/libprotobuf.so  
-- Build type not set - defaulting to Release
-- 
-- ******** Summary ********
--   CMake version         : 3.5.1
--   CMake command         : /usr/bin/cmake
--   System                : Linux
--   C++ compiler          : /usr/bin/c++
--   C++ compiler version  : 5.4.0
--   CXX flags             :  -Wall -Wnon-virtual-dtor
--   Build type            : Release
--   Compile definitions   : ONNX_NAMESPACE=onnx2trt_onnx
--   CMAKE_PREFIX_PATH     : 
--   CMAKE_INSTALL_PREFIX  : /usr/local
--   CMAKE_MODULE_PATH     : 
-- 
--   ONNX version          : 1.3.0
--   ONNX NAMESPACE        : onnx2trt_onnx
--   ONNX_BUILD_TESTS      : OFF
--   ONNX_BUILD_BENCHMARKS : OFF
--   ONNX_USE_LITE_PROTO   : OFF
--   ONNXIFI_DUMMY_BACKEND : OFF
-- 
--   Protobuf compiler     : /usr/local/bin/protoc
--   Protobuf includes     : /usr/local/include
--   Protobuf libraries    : optimized;/usr/local/lib/libprotobuf.so;debug;/usr/local/lib/libprotobuf.so;-lpthread
--   BUILD_ONNX_PYTHON     : OFF
-- Found CUDNN: /usr/include  
-- Found TensorRT headers at /usr/include/aarch64-linux-gnu
-- Find TensorRT libs at /usr/lib/aarch64-linux-gnu/libnvinfer.so;/usr/lib/aarch64-linux-gnu/libnvinfer_plugin.so
-- Found TENSORRT: /usr/include/aarch64-linux-gnu  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/nvidia/onnx-tensorrt/build
----------------------------------------------------------------------------------------------

But when I run the make command, it goes wrong as following:
------------------------------------------------------------------------------------------------
-- ******** Summary ********
--   CMake version         : 3.5.1
--   CMake command         : /usr/bin/cmake
--   System                : Linux
--   C++ compiler          : /usr/bin/c++
--   C++ compiler version  : 5.4.0
--   CXX flags             :  -Wall -Wnon-virtual-dtor
--   Build type            : Release
--   Compile definitions   : ONNX_NAMESPACE=onnx2trt_onnx
--   CMAKE_PREFIX_PATH     : 
--   CMAKE_INSTALL_PREFIX  : /usr/local
--   CMAKE_MODULE_PATH     : 
-- 
--   ONNX version          : 1.3.0
--   ONNX NAMESPACE        : onnx2trt_onnx
--   ONNX_BUILD_TESTS      : OFF
--   ONNX_BUILD_BENCHMARKS : OFF
--   ONNX_USE_LITE_PROTO   : OFF
--   ONNXIFI_DUMMY_BACKEND : OFF
-- 
--   Protobuf compiler     : /usr/local/bin/protoc
--   Protobuf includes     : /usr/local/include
--   Protobuf libraries    : optimized;/usr/local/lib/libprotobuf.so;debug;/usr/local/lib/libprotobuf.so;-lpthread
--   BUILD_ONNX_PYTHON     : OFF
-- Found TensorRT headers at /usr/include/aarch64-linux-gnu
-- Find TensorRT libs at /usr/lib/aarch64-linux-gnu/libnvinfer.so;/usr/lib/aarch64-linux-gnu/libnvinfer_plugin.so
-- Configuring done
-- Generating done
-- Build files have been written to: /home/nvidia/onnx-tensorrt/build
make[2]: Warning: File '../Split.cu' has modification time 28171 s in the future
[  2%] Building NVCC (Device) object CMakeFiles/nvonnxparser_plugin.dir/nvonnxparser_plugin_generated_Split.cu.o
Scanning dependencies of target gen_onnx_proto
make[2]: Warning: File '../third_party/onnx/onnx/onnx.in.proto' has modification time 28176 s in the future
[  4%] Building NVCC (Device) object CMakeFiles/nvonnxparser_plugin.dir/nvonnxparser_plugin_generated_FancyActivation.cu.o
[  6%] Building NVCC (Device) object CMakeFiles/nvonnxparser_plugin.dir/nvonnxparser_plugin_generated_ResizeNearest.cu.o
[  9%] Running gen_proto.py on onnx/onnx.in.proto
Processing /home/nvidia/onnx-tensorrt/third_party/onnx/onnx/onnx.in.proto
Writing /home/nvidia/onnx-tensorrt/build/third_party/onnx/onnx/onnx_onnx2trt_onnx.proto
Writing /home/nvidia/onnx-tensorrt/build/third_party/onnx/onnx/onnx_onnx2trt_onnx.proto3
Writing /home/nvidia/onnx-tensorrt/build/third_party/onnx/onnx/onnx.pb.h
generating /home/nvidia/onnx-tensorrt/build/third_party/onnx/onnx/onnx_pb.py
[ 11%] Running C++ protocol buffer compiler on /home/nvidia/onnx-tensorrt/build/third_party/onnx/onnx/onnx_onnx2trt_onnx.proto
-- Removing /home/nvidia/onnx-tensorrt/build/CMakeFiles/nvonnxparser_plugin.dir//./nvonnxparser_plugin_generated_Split.cu.o
-- Removing /home/nvidia/onnx-tensorrt/build/CMakeFiles/nvonnxparser_plugin.dir//./nvonnxparser_plugin_generated_ResizeNearest.cu.o
-- Removing /home/nvidia/onnx-tensorrt/build/CMakeFiles/nvonnxparser_plugin.dir//./nvonnxparser_plugin_generated_FancyActivation.cu.o
/usr/bin/cmake -E remove /home/nvidia/onnx-tensorrt/build/CMakeFiles/nvonnxparser_plugin.dir//./nvonnxparser_plugin_generated_Split.cu.o
/usr/bin/cmake -E remove /home/nvidia/onnx-tensorrt/build/CMakeFiles/nvonnxparser_plugin.dir//./nvonnxparser_plugin_generated_ResizeNearest.cu.o
/usr/bin/cmake -E remove /home/nvidia/onnx-tensorrt/build/CMakeFiles/nvonnxparser_plugin.dir//./nvonnxparser_plugin_generated_FancyActivation.cu.o
make[2]: warning:  Clock skew detected.  Your build may be incomplete.
[ 11%] Built target gen_onnx_proto
make[2]: Warning: File '../third_party/onnx/onnx/onnx-operators.in.proto' has modification time 28176 s in the future
-- Generating dependency file: /home/nvidia/onnx-tensorrt/build/CMakeFiles/nvonnxparser_plugin.dir//nvonnxparser_plugin_generated_Split.cu.o.NVCC-depend
-- Generating dependency file: /home/nvidia/onnx-tensorrt/build/CMakeFiles/nvonnxparser_plugin.dir//nvonnxparser_plugin_generated_ResizeNearest.cu.o.NVCC-depend
-- Generating dependency file: /home/nvidia/onnx-tensorrt/build/CMakeFiles/nvonnxparser_plugin.dir//nvonnxparser_plugin_generated_FancyActivation.cu.o.NVCC-depend
[ 13%] Running gen_proto.py on onnx/onnx-operators.in.proto
[ 16%] Running gen_proto.py on onnx/onnx.in.proto
/usr/local/cuda/bin/nvcc -M -D__CUDACC__ /home/nvidia/onnx-tensorrt/Split.cu -o /home/nvidia/onnx-tensorrt/build/CMakeFiles/nvonnxparser_plugin.dir//nvonnxparser_plugin_generated_Split.cu.o.NVCC-depend -ccbin /usr/bin/c++ -m64 -DONNX_NAMESPACE=onnx2trt_onnx -Xcompiler ,\"-Wall\",\"-O3\",\"-DNDEBUG\" -cudart static -lineinfo -g --expt-extended-lambda -gencode arch=compute_35,code=sm_35 -gencode arch=compute_53,code=sm_53 -gencode arch=compute_61,code=sm_61 -gencode arch=compute_70,code=sm_70 -gencode arch=compute_75,code=sm_75 -gencode arch=compute_75,code=compute_75 -std=c++11 -Xcompiler -fPIC --expt-extended-lambda -DNVCC -I/usr/include -I/usr/include/aarch64-linux-gnu -I/usr/local/cuda/include -I/usr/local/cuda/include
nvcc fatal   : Unsupported gpu architecture 'compute_75'
CMake Error at nvonnxparser_plugin_generated_Split.cu.o.cmake:207 (message):
  Error generating
  /home/nvidia/onnx-tensorrt/build/CMakeFiles/nvonnxparser_plugin.dir//./nvonnxparser_plugin_generated_Split.cu.o
...
---------------------------------------------------------------------------------------------------
I am grateful if anybody could give some suggestions. The hardware is Drive PX2.

**Comment by RizhaoCai** [comment]
Yes, when I install it, I also get the same error. Do you solve it?

**Comment by Morgan-117** [comment]
@RizhaoCai 
I have modified the `CMakeLists.txt` by remove the 'GPU_ARCHS 70 75', now it looks like:

```
list(APPEND GPU_ARCHS
    35
    53
    61
    )
```
it works on my PX2.(if error "fatal error: api_utils.h: No such file or directory
compilation terminated." appeares, just comment the include code)

**Comment by kevinch-nv** [comment]
In later versions of the onnx-tensorrt parser we've removed dependency on CUDA and have moved all CUDA plugin code to the main [TensorRT repo](https://github.com/NVIDIA/TensorRT/tree/master/plugin). Closing this issue as it no longer affects the latest builds.

For those who still need to build on older branches,  make sure that the `GPU_ARCHS` variable in `CMakeLists.txt` doesn't list a version that your local GPU/CUDA installation doesn't support. For more information, see [here](https://developer.nvidia.com/cuda-gpus)
